packages/authenticated-avl-tree/tests/batch-avl-prover.test.ts

```diff
@@ -0,0 +1,807 @@
+import { BatchAVLProver, AVLTree, InsertOperation, RemoveOperation } from "..";
+
+function hexToBytes(hex: string) {
+  return new Uint8Array(Buffer.from(hex, "hex"));
+}
+
+function bytesToHex(arr: Uint8Array): string {
+  return Buffer.from(arr).toString("hex");
+}
+
+function insertOp(key: string, value: string) {
+  return new InsertOperation(hexToBytes(key), hexToBytes(value));
+}
+
+function removeOp(key: string) {
+  return new RemoveOperation(hexToBytes(key));
+}
+
+describe("BatchAVLProver", () => {
+  describe("Simple operations & proofs", () => {
+    it("should insert and remove", () => {
+      const tree = new AVLTree(32);
+      const bv = new BatchAVLProver(tree, true);
+
+      bv.performOneOperation(
+        new InsertOperation(
+          hexToBytes(
+            "d32c7a04c17b5d34584a92cbca3d9842a5221334d03d1cae608967597e96ef8c"
+          ),
+          hexToBytes("01")
+        )
+      );
+
+      expect(bytesToHex(bv.generateProof())).toBe(
+        "020000000000000000000000000000000000000000000000000000000000000000ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff0000000004"
+      );
+
+      bv.performOneOperation(
+        new RemoveOperation(
+          hexToBytes(
+            "d32c7a04c17b5d34584a92cbca3d9842a5221334d03d1cae608967597e96ef8c"
+          )
+        )
+      );
+
+      expect(bytesToHex(bv.generateProof())).toBe(
+        "020000000000000000000000000000000000000000000000000000000000000000d32c7a04c17b5d34584a92cbca3d9842a5221334d03d1cae608967597e96ef8c0000000002ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff0000000101000400"
+      );
+    });
+  });
+
+  // !! THIS COULD BE THE CAUSE OF BUGS IN THE FUTURE
+  // This test is currently failing when ran by Jest. This is testing
+  // against the set of tests here: https://github.com/knizhnik/scorex_crypto_avltree/blob/36a0b47742d9d8f749ce95d2a794be88a03d13ec/tests/avl_batch_specification.rs#L86
+  // A different proof is being generated after applying `mods` only when running with Jest.
+  // I spent a number of hours debugging but cannot see what could be causing the difference,
+  // if the operations are performed all at once the proofs are the same.
+  // It's only when separated into 2 lists of operations in JS that it starts to fail.
+  // Small separate lists of operations seem to succeed and produce the same proof in jest.
+  //
+  // Both tests should be running inside nodejs with the same WASM code. :shrug:
+  // !! THIS COULD BE THE CAUSE OF BUGS IN THE FUTURE
+  it.skip("should produce the same proofs as the native rust library", () => {
```

Comment:
Will spend a bit more time debugging this but I'm thoroughly confused as to what the cause is.

Added `batch_avl_prover_test.rs` which also tests the WASM in nodejs and passes the test cases, not sure why it fails here

Reply:
When I copy pasted the operation hex strings I made a mistake that cut off a few characters, this was apparently subtle enough to make the first proof generation eq but subsquent proofs differ slightly